text/1001-edgedb-install.rst

```diff
@@ -0,0 +1,298 @@
+..
+    Status: Draft
+    Type: Feature
+    Created: 2020-04-29
+    RFC PR: `edgedb/rfcs#0007 <https://github.com/edgedb/rfcs/pull/7>`_
+
+=================================================================
+RFC 1001: CLI for installation and control of local EdgeDB server
+=================================================================
+
+This RFC describes the design of the ``edgedb`` CLI subcommand for the
+purposes of installation, update and control of a local EdgeDB server.
+
+
+Motivation
+==========
+
+Currently, the tasks of installing, updating and running an EdgeDB server
+instances are entirely manual and vary a lot across the supported platforms.
+From the standpoint of local development this creates unnecessary friction
+and opens lots of possibilities for user error.  The current state also
+necessiates a non-trivial amount of documentation that new users must read.
+
+By implementing the installation, update, and control logic into the ``edgedb``
+CLI we can significantly simplify the process of getting started with EdgeDB
+without leaving the familiar development process:
+
+1. Download the ``edgedb`` binary via the most convenient channel
+   (`npm`, `pip`, `curl`).
+2. Use the downloaded ``edgedb`` binary to either install and initialize
+   a local server, or configure a remote server instance for development.
+
+
+Overview
+========
+
+The RFC proposes a new group of ``edgedb`` CLI commands under ``edgedb server``
+prefix:
+
+* ``edgedb server install`` -- install or update a specific version of the
+  EdgeDB server on the local machine;
+
+* ``edgedb server uninstall`` -- uninstall a specific version of the
+  EdgeDB server or all versions of EdgeDB from the local machine;
+
+* ``edgedb server start`` -- starts an EdgeDB server instance;
+
+* ``edgedb server status`` -- show the status of the local EdgeDB server;
+
+* ``edgedb server stop`` -- stop the given EdgeDB server instance;
+
+* ``edgedb server upgrade`` -- upgrade the specified EdgeDB server instance
+  to the new major version.
+
+* ``edgedb server prune`` -- removes upgrade backups and other unused data.
```

Comment:
There can be at most one backup per instance, so it's more like indicating an availability of a backup rather than "listing" them.  Even then, I doubt it would be all that useful.  The idea is that users do `edgedb server upgrade` and it just works.  The backup and the `--revert` option are there only if things go sideways and you want to quickly recover.